PF4/src/PermutationOfString.java

```diff
@@ -0,0 +1,55 @@
+import java.util.ArrayList;
+import java.util.List;
+/**
+ * 
+ * Class name-PermutationOfString
+ * 
+ * version 1.0
+ * @author Manisha Gupta
+ *
+ */
+
+public class PermutationOfString {
+    ArrayList<String> permutations = new ArrayList<String>();
+    /**
+     * Below function swaps characters at two positions.
+     * @param x
+     * @param i
+     * @param j
+     * @return swapped string
+     */
+    public String swap(String x, int i, int j){
+        char temp;
+        char [] charArray = x.toCharArray();
+        temp = charArray[i];
+        charArray[i] = charArray[j];
+        charArray[j] = temp;
+        return String.valueOf(charArray);
+        
+    }
+    /**
+     * Below function returns all permutations of string using backtracking.
+     * Assuming each character of string is unique.
+     * @param str
+     * @param start
+     * @param end
+     * @return all permutations of string.
+     */
+    public ArrayList<String> generatePermutations(String str, int start, int end){
```

Comment:
Method signature should be same as given in problem. Parameter name should be more logical.